runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/graph/CreateExecutableStageNodeFunction.java

```diff
@@ -191,6 +195,37 @@ public Node apply(MutableNetwork<Node, Edge> input) {
         Iterables.filter(input.nodes(), InstructionOutputNode.class)) {
       InstructionOutput instructionOutput = node.getInstructionOutput();
 
+      // If this is the input PCollection or the output PCollection for an ExecutableStage, it's
+      // necessary to check whether the window coder is not a GlobalWindow coder.
+      if (isExecutableStageInputPCollection(input, node)
+          || isExecutableStageOutputPCollection(input, node)) {
+        Coder<?> javaCoder =
+            CloudObjects.coderFromCloudObject(CloudObject.fromSpec(instructionOutput.getCodec()));
+        // For now, Dataflow runner harness only deal with FixedWindow.
+        if (FullWindowedValueCoder.class.isInstance(javaCoder)) {
+          FullWindowedValueCoder<?> windowedValueCoder = (FullWindowedValueCoder<?>) javaCoder;
+          Coder<?> windowCoder = windowedValueCoder.getWindowCoder();
+          if (IntervalWindowCoder.class.isInstance(windowCoder)) {
+            fakeWindowingStrategyId = "generatedFixedWindowingStrategy" + idGenerator.getId();
+            try {
+              // Since the coder is the only needed from a WindowingStrategy, the size field of one
```

Comment:
Is this WindowingStrategy instantiated in the SDK? If not, I'd much rather provide an actual fake strategy that has nothing but the coder set (or perhaps a nonce for the URN) to make sure it doesn't actually get used.

Reply:
Ideally, the WindowingStrategy should be instantiated in the SDK and passed with pipeline proto to runner harness. However, dataflow runner harness doesn't have pipeline proto thus dataflow generates this fake one in order to make it work with fnapi.

Reply:
Yes, I agree. I'm suggesting we create a very fake one, manually setting the coder_id to the right thing (regardless of whether it's understood by Java), rather than a FixedWindows.

Reply:
Could you please explain more about how we can manually set the coder_id to the right thing?

Reply:
Got it. Thanks~

Reply:
If we don't want to lie on the internal PCollection, then it's not possible to build a totally fake WindowingStrategy. For internal PCollections inside one ExecutanleStage, it needs to initialize WindowFn from WindowingStrategy proto.